contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -4052,6 +4114,126 @@ TdsSendTypeDatetimeoffset(FmgrInfo *finfo, Datum value, void *vMetaData)
 	return rc;
 }
 
+int
+TdsSendTypeGeometry(FmgrInfo *finfo, Datum value, void *vMetaData)
+{
+	int			rc = EOF,
+				npoints,
+				len,			/* number of bytes used to store the string. */
+				actualLen;		/* Number of bytes that would be needed to
+								 * store given string in given encoding. */
+	char	   *destBuf, *buf, *itr;
+	
+	TdsColumnMetaData *col = (TdsColumnMetaData *) vMetaData;
+	GSERIALIZED *gser;          // Used to Store the bytes in the Format which is stored in PostGIS
+	
+	gser = (GSERIALIZED *)PG_DETOAST_DATUM(value);
+	npoints = *((int *)gser->data);
+	/*
+	 * Row chunck length expected by the driver is:
+	 * 16 * (No. of Points) + 6
+	 * 16 -> 2 8-Byte float coordinates (TODO: Need to change when Z and M flags are defined for N-dimension Points)
+	 * 6 -> 4 Byte SRID + 2 Byte (01 0C)
+	*/
+	len = npoints*16 + 6;      
+	buf = (char *) palloc0(len);
+
+	//SRID is stored in reverse order
```

Comment:
is there any risk here with endianness?

Reply:
In T-SQL we have Spatial Data Types Stored always in Little endian order, so, we have used preprocessing to make sure that it remains in order, So, there shouldn't be any risk related to endianness.